src/search/landmarks/landmark_factory_h_m.cc

```diff
@@ -927,8 +929,13 @@ void LandmarkFactoryHM::add_lm_node(int set_index, bool goal) {
         vector<FactPair> facts(hm_entry.fluents);
         utils::sort_unique(facts);
         assert(!facts.empty());
-        Landmark landmark(facts, false, (facts.size() > 1), goal);
-        landmark.first_achievers.insert(
+        std::shared_ptr<Landmark> landmark;
+        if ((facts.size() > 1)){
```

Comment:
I would make the if condition `facts.size() == 1` and thus construct `SimpleLandmark`s there.